transport/x/redis/outbound.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package redis
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/atomic"
+	"go.uber.org/yarpc/internal/errors"
+	"go.uber.org/yarpc/serialize"
+	"go.uber.org/yarpc/transport"
+
+	"github.com/opentracing/opentracing-go"
+)
+
+var errOutboundNotStarted = errors.ErrOutboundNotStarted("redis.Outbound")
+
+// Outbound is a redis OnewayOutbound
+type Outbound struct {
+	client   Client
+	tracer   opentracing.Tracer
+	queueKey string
+
+	started *atomic.Bool
+}
+
+// NewOnewayOutbound creates a redis transport.OnewayOutbound
+func NewOnewayOutbound(client Client, queueKey string) *Outbound {
+	return &Outbound{
+		client:  client,
+		tracer:  opentracing.GlobalTracer(),
+		started: atomic.NewBool(false),
+	}
+}
+
+// WithTracer configures a tracer for the outbound
+func (o *Outbound) WithTracer(tracer opentracing.Tracer) *Outbound {
+	o.tracer = tracer
+	return o
+}
+
+// Start creates connection to the redis instance
+func (o *Outbound) Start() error {
+	var err error
+	if !o.started.Swap(true) {
+		err = o.client.Start()
+	}
+	return err
+}
+
+// Stop stops the redis connection
+func (o *Outbound) Stop() error {
+	var err error
+	if o.started.Swap(false) {
+		err = o.client.Stop()
+	}
+	return err
+}
+
+// CallOneway makes a oneway request using redis
+func (o *Outbound) CallOneway(ctx context.Context, req *transport.Request) (transport.Ack, error) {
+	if !o.started.Load() {
+		panic(errOutboundNotStarted)
+	}
+
+	ctx, span := transport.CreateOpenTracingSpan(ctx, req, o.tracer, "redis", time.Now())
+	defer span.Finish()
+
+	marshalledRPC, err := serialize.ToBytes(o.tracer, span.Context(), req)
+	if err != nil {
+		return nil, transport.UpdateSpanWithErr(span, err)
+	}
+
+	err = o.client.LPush(o.queueKey, marshalledRPC)
+	ack := time.Now()
+
+	if err != nil {
```

Comment:
can we do the error check before we create the `ack` object?

and in that case we can just call
```
return time.Now(), nil
```
at the end.